src/main/resources/com/tikal/jenkins/plugins/multijob/MultiJobTestResults/show-hide-failure-summary.js

```diff
@@ -0,0 +1,55 @@
+const PREFIX = "test-";
+const SHOWLINK_SUFFIX = "-showlink";
+const HIDELINK_SUFFIX = "-hidelink";
+
+function showFailureSummary(summaryId, query) {
+    let element = document.getElementById(summaryId);
+
+    element.style.display = "";
+    document.getElementById(summaryId + SHOWLINK_SUFFIX).style.display = "none";
+    document.getElementById(summaryId + HIDELINK_SUFFIX).style.display = "";
+
+    if (typeof query !== 'undefined' && element.innerHTML.trim() === 'Loading...') {
+        let rqo = new XMLHttpRequest();
+        rqo.open('GET', query, true);
+        rqo.onreadystatechange = function() {
+            element.innerHTML = rqo.responseText;
+            initializeShowHideLinks(element);
+        }
+        rqo.send(null);
+    }
+
+}
+
+function hideFailureSummary(summaryId) {
+    document.getElementById(summaryId).style.display = "none";
+    document.getElementById(summaryId + SHOWLINK_SUFFIX).style.display = "";
+    document.getElementById(summaryId + HIDELINK_SUFFIX).style.display = "none";
+}
+
+function initializeShowHideLinks(container) {
+    container = container || document;
+
+    container.querySelectorAll('a[id$="-showlink"], a[id$="-hidelink"]').forEach(link => {
+        link.addEventListener('click', handleShowHideClick);
+        link.style.cursor = 'pointer';
+    });
+}
+
+function handleShowHideClick(event) {
+    event.preventDefault();
+
+    let link = event.target.closest('a[id$="-showlink"], a[id$="-hidelink"]');
+    const id = link.id.replace(/-showlink$/, '').replace(/-hidelink$/, '');
+    const queryUrl = link.getAttribute('query-url-data');
```

Comment:
```suggestion
    const queryUrl = link.dataset.queryUrl;
```

https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes